File: backend/app/schemas/schedule.py
"""日程相关的Pydantic schemas"""
from pydantic import BaseModel, Field
from typing import Optional
from datetime import date as date_type, time as time_type, datetime


class ScheduleCreateRequest(BaseModel):
    """创建日程请求"""
    date: Optional[str] = Field(default=None, description="日期 (YYYY-MM-DD)")
    time: Optional[str] = Field(default=None, description="时间 (HH:MM)")
    description: str = Field(..., min_length=1, description="事件描述")
    original_text: str = Field(..., min_length=1, description="原始文本")
    
    class Config:
        json_schema_extra = {
            "example": {
                "date": "2024-01-15",
                "time": "14:30",
                "description": "团队会议",
                "original_text": "明天下午2点30分团队会议"
            }
        }


class ScheduleResponse(BaseModel):
    """日程响应"""
    id: str = Field(..., description="日程ID")
    user_id: str = Field(..., description="用户ID")
    date: Optional[date_type] = Field(default=None, description="日期")
    time: Optional[time_type] = Field(default=None, description="时间")
    description: str = Field(..., description="事件描述")
    original_text: str = Field(..., description="原始文本")
    created_at: datetime = Field(..., description="创建时间")
    updated_at: datetime = Field(..., description="更新时间")
    
    class Config:
        from_attributes = True


class ScheduleUpdateRequest(BaseModel):
    """更新日程请求"""
    date: Optional[str] = Field(default=None, description="日期 (YYYY-MM-DD)")
    time: Optional[str] = Field(default=None, description="时间 (HH:MM)")
    description: Optional[str] = Field(default=None, min_length=1, description="事件描述")
    original_text: Optional[str] = Field(default=None, min_length=1, description="原始文本")


class ScheduleListResponse(BaseModel):
    """日程列表响应"""
    schedules: list[ScheduleResponse] = Field(..., description="日程列表")
    total: int = Field(..., description="总数")

File: frontend/src/api/schedule.ts
import apiClient from './client'

export interface Schedule {
  id: string
  user_id: string
  date: string | null
  time: string | null
  description: string
  original_text: string
  created_at: string
  updated_at: string
}

export interface ScheduleCreateRequest {
  date?: string
  time?: string
  description: string
  original_text: string
}

export interface ScheduleUpdateRequest {
  date?: string
  time?: string
  description?: string
  original_text?: string
}

export interface ScheduleListResponse {
  schedules: Schedule[]
  total: number
}

export const scheduleApi = {
  // 创建日程
  async create(data: ScheduleCreateRequest): Promise<Schedule> {
    const response = await apiClient.post('/schedules', data)
    return response.data
  },

  // 获取日程列表
  async list(startDate?: string, endDate?: string): Promise<ScheduleListResponse> {
    const params: any = {}
    if (startDate) params.start_date = startDate
    if (endDate) params.end_date = endDate
    const response = await apiClient.get('/schedules', { params })
    return response.data
  },

  // 获取单个日程
  async get(id: string): Promise<Schedule> {
    const response = await apiClient.get(`/schedules/${id}`)
    return response.data
  },

  // 更新日程
  async update(id: string, data: ScheduleUpdateRequest): Promise<Schedule> {
    const response = await apiClient.put(`/schedules/${id}`, data)
    return response.data
  },

  // 删除日程
  async delete(id: string): Promise<void> {
    await apiClient.delete(`/schedules/${id}`)
  },

  // 格式化日期时间显示
  formatDateTime(date: string | null, time: string | null): string {
    if (!date && !time) return '未设置时间'
    if (date && time) return `${date} ${time}`
    if (date) return date
    if (time) return time
    return ''
  }
}
